tests/PHPUnit/Integration/SqlTest.php

```diff
@@ -43,6 +43,10 @@ public function testOptimize()
         $this->assertTrue(Db::optimizeTables(array('table1', 'table2', 'table3', 'table4')) !== false);
 
         // make sure innodb tables are skipped
-        $this->assertTrue(Db::optimizeTables(array('table3', 'table4')) === false);
+        if (DB::isOptimizeInnoDBSupported()) {
+            $this->assertTrue(Db::optimizeTables(array('table3', 'table4')) !== false);
+        } else {
+            $this->assertTrue(Db::optimizeTables(array('table3', 'table4')) === false);
+        }
```

Comment:
Guess using the proper assertion methods would be better:
```suggestion
        if (DB::isOptimizeInnoDBSupported()) {
            $this->assertTrue(Db::optimizeTables(array('table3', 'table4')));
        } else {
            $this->assertFalse(Db::optimizeTables(array('table3', 'table4')));
        }
```